nestkernel/archiving_node.cpp

```diff
@@ -459,4 +459,236 @@ nest::Archiving_Node::connect_synaptic_element( Name name, int n )
+  , A_LTD_const_( n.A_LTD_const_ )
+  , delay_u_bars_( n.delay_u_bars_ )
+  , ltd_hist_len_( n.ltd_hist_len_ )
+  , ltd_hist_current_( n.ltd_hist_current_ )
+{
+}
+
+void
+nest::Clopath_Archiving_Node::init_clopath_buffers()
+{
+  // Implementation of the delay of the convolved membrane potentials.
+  // This delay is not described in Clopath et al. 2010 but is present in
+  // the code which was presumably used to create the figures in the paper.
+  delayed_u_bars_idx_ = 0;
+  delay_u_bars_steps_ = Time::delay_ms_to_steps( delay_u_bars_ ) + 1;
+  delayed_u_bar_plus_.resize( delay_u_bars_steps_ );
+  delayed_u_bar_minus_.resize( delay_u_bars_steps_ );
+
+  // initialize the ltp-history
+  ltd_hist_current_ = 0;
+  ltd_hist_len_ = kernel().connection_manager.get_max_delay();
+  ltd_history_.resize( ltd_hist_len_, histentry_cl( 0.0, 0.0, 0 ) );
+}
+
+void
+nest::Clopath_Archiving_Node::get_status( DictionaryDatum& d ) const
+{
+  Archiving_Node::get_status( d );
+
+  def< double >( d, names::A_LTD, A_LTD_ );
+  def< double >( d, names::A_LTP, A_LTP_ );
+  def< double >( d, names::u_ref_squared, u_ref_squared_ );
+  def< double >( d, names::theta_plus, theta_plus_ );
+  def< double >( d, names::theta_minus, theta_minus_ );
+  def< bool >( d, names::A_LTD_const, A_LTD_const_ );
+  def< double >( d, names::delay_u_bars, delay_u_bars_ );
+}
+
+void
+nest::Clopath_Archiving_Node::set_status( const DictionaryDatum& d )
+{
+  Archiving_Node::set_status( d );
+
+  // We need to preserve values in case invalid values are set
+  double new_A_LTD = A_LTD_;
+  double new_A_LTP = A_LTP_;
+  double new_theta_plus = theta_plus_;
+  double new_theta_minus = theta_minus_;
+  double new_u_ref_squared = u_ref_squared_;
+  double new_A_LTD_const = A_LTD_const_;
+  double new_delay_u_bars = delay_u_bars_;
+  updateValue< double >( d, names::A_LTD, new_A_LTD );
+  updateValue< double >( d, names::A_LTP, new_A_LTP );
+  updateValue< double >( d, names::u_ref_squared, new_u_ref_squared );
+  updateValue< double >( d, names::theta_plus, new_theta_plus );
+  updateValue< double >( d, names::theta_minus, new_theta_minus );
+  updateValue< bool >( d, names::A_LTD_const, new_A_LTD_const );
+  updateValue< double >( d, names::delay_u_bars, new_delay_u_bars );
+  A_LTD_ = new_A_LTD;
+  A_LTP_ = new_A_LTP;
+  u_ref_squared_ = new_u_ref_squared;
+
+  if ( u_ref_squared_ <= 0 )
+  {
+    throw BadProperty( "Ensure that u_ref_squared > 0" );
+  }
+
+  theta_plus_ = new_theta_plus;
+  theta_minus_ = new_theta_minus;
+  A_LTD_const_ = new_A_LTD_const;
+  delay_u_bars_ = new_delay_u_bars;
+}
+
+double
+nest::Clopath_Archiving_Node::get_LTD_value( double t )
+{
+  std::vector< histentry_cl >::iterator runner;
+  if ( ltd_history_.empty() || t < 0.0 )
+  {
+    // Return 0.0 if history is empty
+    return 0.0;
+  }
+  else
+  {
+    runner = ltd_history_.begin();
+    while ( runner != ltd_history_.end() )
+    {
+      if ( abs( t - runner->t_ ) < 1e-4 )
```

Comment:
Can we make this comparison more elegant? Are we expecting these times to be always the same to within floating point tolerance of the machine?

Reply:
Changed abs() to fabs() and replaced 1e-4 by 1.0e-7. Yes, these times are always the same but only within floating point tolerance. Therefore, we cannot check for equality.

Reply:
@clinssen We were not aware of the existence of `kernel().connection_manager.get_stdp_eps()`, but it is exactly what is needed here, thanks for the suggestion. Changed with 4a276e2.